### PR TITLE
Add public methods for JSON serde

### PR DESCRIPTION
**Pull request by avantgardnerio**
1. Add public methods for `to_json` and `from_json`
2. change the `serde` feature name to `json` (which seems more accurate/intuitive)

### CODE REVIEW

File: datafusion/proto/src/bytes/mod.rs [hunk 1]
```diff
@@ -107,6 +107,19 @@ pub fn logical_plan_to_bytes(plan: &LogicalPlan) -> Result<Bytes> {
     logical_plan_to_bytes_with_extension_codec(plan, &extension_codec)
 }
 
+/// Serialize a LogicalPlan as json
+#[cfg(feature = "json")]
```
Comment (avantgardnerio) [inline datafusion/proto/src/bytes/mod.rs:111]: The only weird part about this PR is that these methods just disappear if the `json` feature is disabled - would it be better to put them in another file or something?

File: datafusion/proto/Cargo.toml [hunk 2]
```diff
@@ -34,7 +34,7 @@ path = "src/lib.rs"
 
 [features]
 default = []
-serde = ["pbjson", "pbjson-build", "dep:serde"]
+json = ["pbjson", "pbjson-build", "dep:serde"]
```
Comment (tustvold) [inline datafusion/proto/Cargo.toml:37]:
```suggestion
json = ["pbjson", "pbjson-build", "serde"]
```

File: datafusion/proto/src/bytes/mod.rs (hunk 1 above)
Comment (tustvold) [reply datafusion/proto/src/bytes/mod.rs:111]: I see no issue with this, it is fairly normal across the ecosystem

**Review by tustvold — COMMENTED** [review]
Only 1 minor nit

**Pull request merged**